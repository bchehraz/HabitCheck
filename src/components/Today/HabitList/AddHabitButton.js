import React, { useState, useRef } from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import { FaPlus } from "react-icons/fa"
import { BsCheck } from "react-icons/bs"

import Backdrop from "../../Backdrop"

const Container = styled.div`
  position: relative;
  margin: 0;
  padding: 0;
  width: 100%;
  height: ${({ formOpen }) => (formOpen ? "180px" : "53px")};

  display: flex;
  flex-flow: column nowrap;
  justify-content: center;
  align-items: center;

  border: 2px solid #2d3142;
  border-radius: 17px;
  background-color: #2d3142;
  color: white;

  transition: all 300ms ease-out;
  z-index: 10;

  @media only screen and (min-width: 450px) {
    height: ${({ formOpen }) => (formOpen ? "119px" : "53px")};
  }

  @media only screen and (min-width: 768px) {
    height: 119px;
    box-shadow: 3px 3px 7px rgba(0, 0, 0, 0.29);
    background: linear-gradient(97.52deg, #66c7f4 0%, #8acb88 100%);
    border: none;
  }

  @media only screen and (min-width: 1024px) {
    height: 135px;
  }

  @media only screen and (min-width: 1440px) {
    height: 150px;
  }
`

const IconContainer = styled.div`
  position: absolute;
  display: flex;
  flex-flow: row nowrap;
  justify-content: center;
  align-items: center;

  transition: all 300ms ease-out;
  opacity: ${props => (props.formOpen ? 0 : 1)};

  .icon {
    width: 25px;
    height: 25px;
  }

  @media only screen and (min-width: 768px) {
    .icon {
      width: 40px;
      height: 40px;
    }
  }
`

const FormContainer = styled.form`
  display: flex;
  flex-flow: column nowrap;
  z-index: 10;
  width: 100%;
  padding: 1rem;

  transition: all 150ms ease-out;
  margin: 0 auto;

  transform: translateY(75px);

  opacity: 0;
  visibility: hidden;

  ${props =>
    props.formOpen &&
    `
    visibility: visible;
    opacity: 1;
    transform: translateY(0);
    transform: translateX(0);
    transition: all 300ms ease-out;
  `}

  @media only screen and (min-width: 768px) {
    transform: translateY(0);
    ${props =>
      props.formOpen &&
      `
      visibility: visible;
      opacity: 1;
      transition: all 300ms ease-out;
    `}
  }
`

const FormSection = styled.div`
  display: flex;
  flex-flow: column nowrap;
  margin-top: 10px;

  @media only screen and (min-width: 450px) {
    flex-flow: row nowrap;
    justify-content: center;
    align-items: center;
  }
`

const TextInput = styled.input`
  padding: 10px 5px;
  width: 100%;
  background-color: #6f789d;
  color: white;
  border-bottom: 2px solid #8acb88;

  ${({ error }) =>
    error &&
    `
    border-bottom: 3px solid #AA4465;
  `}

  @media only screen and (min-width: 450px) {
    padding: 10px 10px;
    color: #2d3142;
    font-weight: 600;
  }

  @media only screen and (min-width: 768px) {
    font-size: 1.2em;
    background-color: white;
  }
`

const SubmitButton = styled.button`
  color: white;
  background-color: #8acb88;
  width: 100%;
  margin-top: 10px;
  padding: 5px;

  .buttonIcon {
    width: 30px;
    height: 30px;
  }

  @media only screen and (min-width: 450px) {
    width: auto;
    padding: 0;
    margin-top: 0;
    margin-left: 10px;
    background-color: rgba(0, 0, 0, 0);

    .buttonIcon {
      width: 40px;
      height: 100%;
      color: #8acb88;
    }
  }

  @media only screen and (min-width: 768px) {
    .buttonIcon {
      color: #fff;
    }
  }
`

const Header = styled.h4`
  color: white;
  margin: 0;
  text-align: left;
`

const validForm = formInput => {
  const { title } = formInput
  return title && title !== "" && title.length <= 64
}

const AddHabitButton = ({ handleAddHabit, onNewHabitAdded }) => {
  const initState = {
    title: "",
  }

  const [formOpen, setFormOpen] = useState(false)
  const [formInput, setFormInput] = useState(initState)
  const [error, setError] = useState(false)
  const textInput = useRef(null)

  const onFormOpen = () => {
    if (!formOpen) {
      setFormOpen(true)

      // Due to Form goes from hidden to visible,
      // input focus() won't work if the function is run immediately after
      // this is the solution for the time being
      window.setTimeout(() => {
        textInput.current.focus()
      }, 300)
    }
  }

  const onFormClose = () => {
    if (formOpen) {
      setFormOpen(!formOpen)
      setError(false)
      setFormInput(initState)
    }
  }

  const onTextChange = e => {
    const { value } = e.target
    setError(!(value.length <= 64))
    setFormInput({ ...formInput, title: e.target.value })
  }

  return (
    <div>
      <Container
        className="addHabitButton"
        formOpen={formOpen}
        onClick={onFormOpen}
      >
        <FormContainer formOpen={formOpen} className="form">
          <Header formOpen={formOpen}>{"New Habit"}</Header>

          <FormSection>
            <TextInput
              className="habitTitleInput"
              type="text"
              value={formInput.title}
              onChange={onTextChange}
              error={error}
              ref={textInput}
            />

            <SubmitButton
              onClick={e => {
                e.preventDefault()
                if (validForm(formInput)) {
                  handleAddHabit(formInput.title)
                  setError(false)
                  setFormOpen(false)
                  setFormInput(initState)
                  onNewHabitAdded()
                } else {
                  setError(true)
                }
              }}
            >
              <BsCheck className="buttonIcon" />
            </SubmitButton>
          </FormSection>
        </FormContainer>

        <IconContainer
          formOpen={formOpen}
          className={(formOpen && "iconContainerClose") || "iconContainerOpen"}
          data-testid="AddHabitButtonClose"
        >
          <FaPlus className="icon" />
        </IconContainer>
      </Container>
      <Backdrop onClick={onFormClose} enabled={formOpen} zIndex={9} />
    </div>
  )
}

AddHabitButton.propTypes = {
  handleAddHabit: PropTypes.func.isRequired,
  onNewHabitAdded: PropTypes.func.isRequired,
}

export default AddHabitButton
