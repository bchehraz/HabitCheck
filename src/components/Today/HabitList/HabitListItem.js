import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"

import { FaCheck, FaSquare } from "react-icons/fa"

import { Tag } from "./"

const ItemContainer = styled.div`
  margin-bottom: 14px;
  padding: 20px;
  width: 327px;
  height: 53px;

  display: flex;
  flex-flow: row nowrap;
  justify-content: space-between;
  align-items: center;

  border: 2px solid #2D3142;
  ${({ loseStreak }) => loseStreak && "border-color: #AA4465;"}
  ${({ checked }) => checked && "border-color: #8ACB88;"}
  ${({ checked }) => checked && "background-color: #8ACB88;"}
  ${({ checked }) => checked && "color: #FFFFFF;"}
  border-radius: 17px;

  h4 {
    ${({ checked }) => checked && "color: #FFFFFF;"}
  }

  position: relative;
`

const LeftClickable = styled.div`
  position: absolute;
  width: 50%;
  height: 100%;
  left: 0%;
  cursor: pointer;
`

const RightClickable = styled.div`
  position: absolute;
  width: 25%;
  height: 100%;
  left: 75%;
  cursor: pointer;
`

const Status = styled.div`
  display: flex;
  flex-direction: row nowrap;
  justify-content: flex-end;
  align-items: center;
`

const CheckBox = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  padding-left: 10px;
`

const HabitListItem = ({
  title,
  streak,
  isNew,
  isChecked,
  onClick,
  viewStats,
}) => {
  return (
    <ItemContainer
      data-testid="HabitListItem"
      checked={isChecked}
      loseStreak={streak < 0}
    >
      <LeftClickable onClick={viewStats} className="toHabitStats" />
      <RightClickable onClick={onClick} className="habitListItemAction" />

      <h4 style={{ margin: 0 }}>{title}</h4>

      <Status className="habitListItemStatus">
        <Tag streak={streak} isNew={isNew} />
        <CheckBox className="habitListItemCheckbox">
          {(isChecked && <FaCheck size={23} />) || (
            <FaSquare
              size={23}
              style={{
                color: "white",
                border: "1px solid #2D3142",
                borderRadius: "4px",
              }}
            />
          )}
        </CheckBox>
      </Status>
    </ItemContainer>
  )
}

HabitListItem.propTypes = {
  title: PropTypes.string,
  streak: PropTypes.number,
  isNew: PropTypes.bool,
  isChecked: PropTypes.bool,
  onClick: PropTypes.func.isRequired,
  viewStats: PropTypes.func.isRequired,
}

export default HabitListItem
