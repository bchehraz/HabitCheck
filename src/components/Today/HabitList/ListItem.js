import React, { useState, useEffect } from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import { FaSquare } from "react-icons/fa"
import { BsCheck } from "react-icons/bs"

import { Tag } from "./"

const Container = styled.div`
  margin-bottom: 14px;
  display: flex;
  flex-flow: column nowrap;
  align-items: space-between;
  justify-content: center;

  width: 314px;
  width: 100%;
  height: 119px;
  border: none;
  justify-content: flex-start;
  box-shadow: 3px 3px 7px rgba(0, 0, 0, 0.29);
  margin: 0 auto;
  transform: rotateX(-90deg) translateX(-100px);
  opacity: 0;
  user-select: none;

  ${({ streak }) => streak < 0 && "border-color: #AA4465;"}
  ${({ checked }) => checked && "border-color: #8ACB88;"}
  border-radius: 17px;

  ${({ checked }) => checked && "background-color: #8ACB88;"}
  ${({ checked }) => checked && "color: #FFFFFF;"}

  h4 {
    ${({ checked }) => checked && "color: #FFFFFF;"}
    font-weight: 600;
    font-size: 1.2em;
  }

  /* Animation */
  &.show {
    opacity: 1;
    transform: none;
    transition: all 0.5s cubic-bezier(0.36, -0.64, 0.34, 1.76);
  }

  &.animate {
    opacity: 1;
    transform: none;
    transition: all 0.5s cubic-bezier(0.36, -0.64, 0.34, 1.76);
  }

  @media only screen and (min-width: 1024px) {
    height: 135px;
  }

  @media only screen and (min-width: 1440px) {
    height: 150px;
  }
`

const Head = styled.div`
  display: flex;
  flex-flow: row nowrap;
  justify-content: space-between;
  ${({ checked }) => !checked && "justify-content: flex-end;"}
  align-items: center;
  width: 100%;
  ${({ streak }) => streak >= 0 && "background-color: #2D3142;"}
  ${({ streak }) => streak < 0 && "background-color: #AA4465;"}
  ${({ checked }) => checked && "background-color: #5AAF58;"}
  ${({ isNew }) => isNew && "background-color: #66C7F4;"}
  border-radius: 17px 17px 0 0;
  padding: 0 1em;
  height: 42px;

  h4 {
    color: #B6ECB4;
    margin: 0;
  }
`

const Body = styled.div`
  position: relative;
  display: flex;
  flex-flow: row nowrap;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  height: calc(100% - 30px);
  padding: 2em 1em;
  border-radius: 0 0 17px 17px;
  border: 2px solid #2D3142;
  ${({ streak }) => streak < 0 && "border-color: #AA4465;"}
  ${({ checked }) => checked && "border-color: #8ACB88;"}
  ${({ isNew }) => isNew && "border-color: #66C7F4;"}
  border-radius: 0 0 17px 17px;
  transition: all 300ms ease-out;
`

const Status = styled.div`
  display: flex;
  flex-direction: row nowrap;
  justify-content: flex-end;
  align-items: center;
  cursor: pointer;

  padding: 20px 10px;
`

const CheckBox = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;

  .checkbox,
  .check {
    width: 40px;
    height: 40px;
  }
`

const Clickable = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  width: 73%;
  height: 100%;
  cursor: pointer;
`

const ListItem = ({
  title,
  streak,
  isNew,
  isChecked,
  onClick,
  viewStats,
  justAdded,
  toggleAnimate,
}) => {
  const [animated, setAnimated] = useState(false)

  useEffect(() => {
    if (justAdded) {
      setAnimated(true)
      toggleAnimate()
    } else {
      setAnimated(false)
    }
  }, [justAdded])

  return (
    <Container
      className={(!justAdded && "show") || (animated && "animate")}
      data-testid="HabitListItem"
      checked={isChecked}
      streak={streak}
      isNew={isNew}
    >
      <Head checked={isChecked} streak={streak} isNew={isNew}>
        {isChecked && <h4>Completed</h4>}
        <Tag streak={streak} isNew={isNew} tablet />
      </Head>
      <Body streak={streak} checked={isChecked} isNew={isNew}>
        <Clickable onClick={viewStats} />

        <h4 style={{ margin: 0 }}>{title}</h4>

        <Status className="habitListItemStatus" onClick={onClick}>
          <CheckBox className="habitListItemCheckbox">
            {(isChecked && <BsCheck className="check" />) || (
              <FaSquare
                className="checkbox"
                style={{
                  color: "white",
                  border: "1px solid #2D3142",
                  borderRadius: "4px",
                }}
              />
            )}
          </CheckBox>
        </Status>
      </Body>
    </Container>
  )
}

ListItem.propTypes = {
  title: PropTypes.string,
  streak: PropTypes.number,
  isNew: PropTypes.bool,
  isChecked: PropTypes.bool,
  onClick: PropTypes.func.isRequired,
  viewStats: PropTypes.func.isRequired,
  justAdded: PropTypes.bool,
  toggleAnimate: PropTypes.func,
}

ListItem.defaultProps = {
  justAdded: false,
}

export default ListItem
