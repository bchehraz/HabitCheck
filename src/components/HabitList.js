import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"

import HabitListItem from "./HabitListItem"
import AddHabitButton from "./AddHabitButton"

const HabitListContainer = styled.div`
  width: 100%;
`

const HabitList = ({ habits, handleAddHabit, checkHabit, uncheckHabit }) => (
  <HabitListContainer>
    <AddHabitButton handleAddHabit={handleAddHabit} />
    {habits.unchecked.map((habit, index) => {
      const { title } = habit
      let streak = 0
      if (habit.progress.length !== 0) {
        const { length } = habit.progress
        streak = habit.progress[length - 1].streak
      }
      return (
        <HabitListItem
          key={index}
          title={title}
          streak={streak}
          isNew={streak === 0}
          isChecked={false}
          onClick={() => {
            checkHabit(index)
          }}
        />
      )
    })}
    {habits.checked.map((habit, index) => {
      const { length } = habit.progress
      const { title, progress } = habit
      const { streak } = progress[length - 1]

      return (
        <HabitListItem
          key={index}
          title={title}
          streak={streak}
          isNew={streak === 0}
          isChecked={true}
          onClick={() => {
            uncheckHabit(index)
          }}
        />
      )
    })}
  </HabitListContainer>
)

HabitList.propTypes = {
  habits: PropTypes.object.isRequired,
  handleAddHabit: PropTypes.func.isRequired,
}

export default HabitList
