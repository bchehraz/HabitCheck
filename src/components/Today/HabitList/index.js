import Emoji from "./Emoji"
import HabitList from "./HabitList"
import HabitListItem from "./HabitListItem"
import Tag from "./Tag"
import AddHabitButton from "./AddHabitButton"

export { Emoji, HabitList, HabitListItem, Tag, AddHabitButton }
