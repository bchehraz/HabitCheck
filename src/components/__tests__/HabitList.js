import React from "react"
import renderer from "react-test-renderer"

import HabitList from "../HabitList"
import { data_1, data } from "../../utils/data"

describe("HabitList Component Snapshot", () => {
  it("renders with empty data", () => {
    const tree = renderer
      .create(
        <HabitList
          habits={data_1.habits}
          handleAddHabit={() => {}}
          checkHabit={() => {}}
          uncheckHabit={() => {}}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("renders with data", () => {
    const tree = renderer
      .create(
        <HabitList
          habits={data.habits}
          handleAddHabit={() => {}}
          checkHabit={() => {}}
          uncheckHabit={() => {}}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
