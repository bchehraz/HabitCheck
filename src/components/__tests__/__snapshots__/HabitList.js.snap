// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HabitList Component Snapshot renders with data 1`] = `
<div
  className="sc-gqjmRU gggsyL habitList"
>
  <div
    className="sc-gzVnrw kmJYvF addHabitButton"
    onClick={[Function]}
  >
    <form
      className="sc-dnqmqq jgPXaW"
    >
      <input
        className="sc-iwsKbI jOlfmC habitTitleInput"
        onChange={[Function]}
        placeholder="Habit Title"
        type="text"
        value=""
      />
      <button
        className="sc-gZMcBi hiiSes"
        onClick={[Function]}
      >
        Start New Habit
      </button>
    </form>
    <div
      className="sc-htoDjs kUfOZM"
      data-testid="AddHabitButtonClose"
      onClick={[Function]}
    >
      <svg
        className="icon"
        fill="currentColor"
        height={25}
        size={25}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 448 512"
        width={25}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
        />
      </svg>
    </div>
  </div>
  <div
    checked={false}
    className="sc-bdVaJa flVaPf"
    data-testid="HabitListItem"
  >
    <div
      className="sc-ifAKCX eySOSY habitListItemAction"
      onClick={[Function]}
    />
    <h4
      style={
        Object {
          "margin": 0,
        }
      }
    >
      Work on HabitCheck
    </h4>
    <div
      className="sc-EHOje BsGex habitListItemStatus"
    >
      <div
        className="sc-bwzfXH jbgUER"
      >
        <p
          style={
            Object {
              "margin": 0,
            }
          }
        >
          NEW
        </p>
      </div>
      <div
        className="sc-bZQynM gWESGZ habitListItemCheckbox"
      >
        <svg
          fill="currentColor"
          height={23}
          size={23}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "border": "1px solid #2D3142",
              "borderRadius": "4px",
              "color": "white",
            }
          }
          viewBox="0 0 448 512"
          width={23}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48z"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    checked={false}
    className="sc-bdVaJa flVaPf"
    data-testid="HabitListItem"
  >
    <div
      className="sc-ifAKCX eySOSY habitListItemAction"
      onClick={[Function]}
    />
    <h4
      style={
        Object {
          "margin": 0,
        }
      }
    >
      TEST_DATA_1
    </h4>
    <div
      className="sc-EHOje BsGex habitListItemStatus"
    >
      <div
        className="sc-bwzfXH kvljid"
      >
        <div
          className="sc-htpNat itjhXY"
        >
          <p
            className="number"
            style={
              Object {
                "fontSize": 18,
              }
            }
          >
            24
          </p>
          <span
            aria-hidden="false"
            aria-label="fire"
            className="emoji"
            role="img"
            style={
              Object {
                "fontSize": 19,
              }
            }
          >
            ðŸ”¥
          </span>
        </div>
      </div>
      <div
        className="sc-bZQynM gWESGZ habitListItemCheckbox"
      >
        <svg
          fill="currentColor"
          height={23}
          size={23}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "border": "1px solid #2D3142",
              "borderRadius": "4px",
              "color": "white",
            }
          }
          viewBox="0 0 448 512"
          width={23}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48z"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    checked={false}
    className="sc-bdVaJa flVaPf"
    data-testid="HabitListItem"
  >
    <div
      className="sc-ifAKCX eySOSY habitListItemAction"
      onClick={[Function]}
    />
    <h4
      style={
        Object {
          "margin": 0,
        }
      }
    >
      Quit Smoking
    </h4>
    <div
      className="sc-EHOje BsGex habitListItemStatus"
    >
      <div
        className="sc-bwzfXH kvljid"
      >
        <div
          className="sc-htpNat itjhXY"
        >
          <p
            className="number"
            style={
              Object {
                "fontSize": 18,
              }
            }
          >
            40
          </p>
          <span
            aria-hidden="false"
            aria-label="fire"
            className="emoji"
            role="img"
            style={
              Object {
                "fontSize": 19,
              }
            }
          >
            ðŸ”¥
          </span>
        </div>
      </div>
      <div
        className="sc-bZQynM gWESGZ habitListItemCheckbox"
      >
        <svg
          fill="currentColor"
          height={23}
          size={23}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "border": "1px solid #2D3142",
              "borderRadius": "4px",
              "color": "white",
            }
          }
          viewBox="0 0 448 512"
          width={23}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48z"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    checked={false}
    className="sc-bdVaJa bWpfaw"
    data-testid="HabitListItem"
  >
    <div
      className="sc-ifAKCX eySOSY habitListItemAction"
      onClick={[Function]}
    />
    <h4
      style={
        Object {
          "margin": 0,
        }
      }
    >
      Meditate
    </h4>
    <div
      className="sc-EHOje BsGex habitListItemStatus"
    >
      <div
        className="sc-bwzfXH jJYxWb"
      >
        <div
          className="sc-htpNat itjhXY"
        >
          <p
            className="number"
            style={
              Object {
                "fontSize": 18,
              }
            }
          >
            -3
          </p>
          <div
            className="sc-bxivhb dNeoOh"
          >
            <svg
              fill="none"
              height={21}
              size={21}
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width={21}
              xmlns="http://www.w3.org/2000/svg"
            >
              <polyline
                points="23 18 13.5 8.5 8.5 13.5 1 6"
              />
              <polyline
                points="17 18 23 18 23 12"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        className="sc-bZQynM gWESGZ habitListItemCheckbox"
      >
        <svg
          fill="currentColor"
          height={23}
          size={23}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "border": "1px solid #2D3142",
              "borderRadius": "4px",
              "color": "white",
            }
          }
          viewBox="0 0 448 512"
          width={23}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48z"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    checked={false}
    className="sc-bdVaJa bWpfaw"
    data-testid="HabitListItem"
  >
    <div
      className="sc-ifAKCX eySOSY habitListItemAction"
      onClick={[Function]}
    />
    <h4
      style={
        Object {
          "margin": 0,
        }
      }
    >
      Sleep on time
    </h4>
    <div
      className="sc-EHOje BsGex habitListItemStatus"
    >
      <div
        className="sc-bwzfXH jJYxWb"
      >
        <div
          className="sc-htpNat itjhXY"
        >
          <p
            className="number"
            style={
              Object {
                "fontSize": 18,
              }
            }
          >
            -1
          </p>
          <div
            className="sc-bxivhb dNeoOh"
          >
            <svg
              fill="none"
              height={21}
              size={21}
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width={21}
              xmlns="http://www.w3.org/2000/svg"
            >
              <polyline
                points="23 18 13.5 8.5 8.5 13.5 1 6"
              />
              <polyline
                points="17 18 23 18 23 12"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        className="sc-bZQynM gWESGZ habitListItemCheckbox"
      >
        <svg
          fill="currentColor"
          height={23}
          size={23}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "border": "1px solid #2D3142",
              "borderRadius": "4px",
              "color": "white",
            }
          }
          viewBox="0 0 448 512"
          width={23}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48z"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    checked={true}
    className="sc-bdVaJa fJLxhr"
    data-testid="HabitListItem"
  >
    <div
      className="sc-ifAKCX eySOSY habitListItemAction"
      onClick={[Function]}
    />
    <h4
      style={
        Object {
          "margin": 0,
        }
      }
    >
      90 min Yoga
    </h4>
    <div
      className="sc-EHOje BsGex habitListItemStatus"
    >
      <div
        className="sc-bZQynM gWESGZ habitListItemCheckbox"
      >
        <svg
          fill="currentColor"
          height={23}
          size={23}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width={23}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    checked={true}
    className="sc-bdVaJa fJLxhr"
    data-testid="HabitListItem"
  >
    <div
      className="sc-ifAKCX eySOSY habitListItemAction"
      onClick={[Function]}
    />
    <h4
      style={
        Object {
          "margin": 0,
        }
      }
    >
      Push Ups
    </h4>
    <div
      className="sc-EHOje BsGex habitListItemStatus"
    >
      <div
        className="sc-bZQynM gWESGZ habitListItemCheckbox"
      >
        <svg
          fill="currentColor"
          height={23}
          size={23}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width={23}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    checked={true}
    className="sc-bdVaJa fJLxhr"
    data-testid="HabitListItem"
  >
    <div
      className="sc-ifAKCX eySOSY habitListItemAction"
      onClick={[Function]}
    />
    <h4
      style={
        Object {
          "margin": 0,
        }
      }
    >
      30 min walk
    </h4>
    <div
      className="sc-EHOje BsGex habitListItemStatus"
    >
      <div
        className="sc-bZQynM gWESGZ habitListItemCheckbox"
      >
        <svg
          fill="currentColor"
          height={23}
          size={23}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width={23}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    checked={true}
    className="sc-bdVaJa fJLxhr"
    data-testid="HabitListItem"
  >
    <div
      className="sc-ifAKCX eySOSY habitListItemAction"
      onClick={[Function]}
    />
    <h4
      style={
        Object {
          "margin": 0,
        }
      }
    >
      Keto Diet
    </h4>
    <div
      className="sc-EHOje BsGex habitListItemStatus"
    >
      <div
        className="sc-bwzfXH kvljid"
      >
        <div
          className="sc-htpNat itjhXY"
        >
          <p
            className="number"
            style={
              Object {
                "fontSize": 18,
              }
            }
          >
            3
          </p>
          <span
            aria-hidden="false"
            aria-label="fire"
            className="emoji"
            role="img"
            style={
              Object {
                "fontSize": 19,
              }
            }
          >
            ðŸ”¥
          </span>
        </div>
      </div>
      <div
        className="sc-bZQynM gWESGZ habitListItemCheckbox"
      >
        <svg
          fill="currentColor"
          height={23}
          size={23}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width={23}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`HabitList Component Snapshot renders with empty data 1`] = `
<div
  className="sc-gqjmRU gggsyL habitList"
>
  <div
    className="sc-gzVnrw kmJYvF addHabitButton"
    onClick={[Function]}
  >
    <form
      className="sc-dnqmqq jgPXaW"
    >
      <input
        className="sc-iwsKbI jOlfmC habitTitleInput"
        onChange={[Function]}
        placeholder="Habit Title"
        type="text"
        value=""
      />
      <button
        className="sc-gZMcBi hiiSes"
        onClick={[Function]}
      >
        Start New Habit
      </button>
    </form>
    <div
      className="sc-htoDjs kUfOZM"
      data-testid="AddHabitButtonClose"
      onClick={[Function]}
    >
      <svg
        className="icon"
        fill="currentColor"
        height={25}
        size={25}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 448 512"
        width={25}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
        />
      </svg>
    </div>
  </div>
</div>
`;
